### ISSUE TITLE
ArgsTable does not work in MDX when using EmotionCSS

### ISSUE THREAD

**Issue by AshConnolly** [storybookjs/storybook#12970]
**Describe the bug**
I want to use Storybook with MDX file format and have my proptypes displayed in a table. It seems `ArgsTable` does not work in MDX when using EmotionCSS styling.

## [Github repo of the issue](https://github.com/AshConnolly/storybook-emotion-issue)

To enable emotionCSS `storybook/.main,js` needs to include an emotion babel preset like so:
```js
module.exports = {
  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
  addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
  babel: async options => ({
    presets: ['@emotion/babel-preset-css-prop'],
  }),
}
```

Unfortunately this stops the ArgsTable from working. [Heres an image with the above main.js](https://imgur.com/CAqjFPj.png) notice how the components get the styling correctly.

[Heres an image with `presets: ['@emotion/babel-preset-css-prop']` removed](https://imgur.com/3UL9pIT.png) from `.storybook/main.js`, notice how the `ArgsTable` now works, but styling is not working.

Header component:
```js
import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import React from 'react'

const styles = css`
  margin: 10px;   padding: 10px;   font-family: sans-serif;   background: papayawhip;
  a { margin-right: 10px; display: inline-block; }
`

export const Header = props => {
  return (
    <header css={styles}>
      <p>{props.text}</p>
      <a href="/home">Home</a>
      <a href="/api/getShowData?show=">/api/getShowData?show=</a>
      <a href="/api/search?show=">/api/search?show=</a>
    </header>
  )
}

Header.propTypes = { text: PropTypes.string}
Header.defaultProps = {text: 'Default text prop'}
```

Header.stories.mdx:
```js
import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks'
import { Header } from './Header'

<Meta title="Components/Header" component={Header} />

# Header

## Props

<ArgsTable of={Header} />

## Preview

<Canvas>
  <Header />
  <Header text="Lorem Ipsum" />
</Canvas>
```

**To Reproduce**
Have an app with storybook and emotion and use an MDX story with ArgsTable.
See repo here - 

**Expected behavior**
A component in storybook that receives its styling and has a props table visible.

**System**
```
Environment Info:

  System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Browsers:
    Chrome: 86.0.4240.111
    Firefox: 80.0.1
    Safari: 14.0
  npmPackages:
    @storybook/addon-actions: ^6.0.28 => 6.0.28 
    @storybook/addon-essentials: ^6.0.28 => 6.0.28 
    @storybook/addon-links: ^6.0.28 => 6.0.28 
    @storybook/react: ^6.0.28 => 6.0.28 
```

**Comment by mattmoreira** [comment]
@AshConnolly, maybe this will solve your problem?

https://github.com/storybookjs/storybook/issues/7540#issuecomment-689190904

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid. Please open a new issue referencing this one if:
- this is still a problem in SB 7.x
- you can provide a consistent reproduction in 7.x (at https://storybook.new/ or per [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce)).

FWIW, I'm pretty sure this one was fixed in v6.5 or so.
